crates/ruma-federation-api/src/keys/get_keys/v1.rs

```diff
@@ -22,9 +25,23 @@ ruma_api! {
         pub device_keys: BTreeMap<UserId, Vec<DeviceIdBox>>,
     }
 
+    #[derive(Default)]
     response: {
         /// Keys from the queried devices.
+        #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
```

Comment:
Since the `device_keys` field is marked required, we should not skip serializing it even if there are no entries. I don't think we should have the default attribute either.